### PR TITLE
Update MinioClientSettings to UseSsl parsing

### PR DESCRIPTION
**Pull request by josemalm32**
- [x] Create branch fix/minio-usesl-strict from main
- [x] Update MinioClientSettings.cs to make UseSsl parsing strict
  - [x] Add UseSslKeys constant array for UseSsl variants (UseSsl, UseSSL, Ssl)
  - [x] Update ParseConnectionString to handle UseSsl from connection string
  - [x] Add TryParseBool method with strict bool parsing (only "true"/"false")
  - [x] Remove any URI scheme-based UseSsl inference
- [x] Add comprehensive tests for the new strict parsing behavior
  - [x] Test explicit UseSsl=true/false in connection string
  - [x] Test that URI scheme doesn't affect UseSsl
  - [x] Test strict boolean parsing (reject "1"/"0", "yes"/"no", "on"/"off")
  - [x] Test all UseSsl key variants (UseSsl, UseSSL, Ssl)
  - [x] Test case-insensitive parsing
- [x] Address code review feedback
  - [x] Translate Spanish comment to English
  - [x] Improve null handling in TryParseBool method

Changes made:
- Updated ParseConnectionString to no longer infer UseSsl from https/http URI schemes
- Added strict TryParseBool that only accepts boolean values or "true"/"false" strings
- Added support for UseSsl key variants (UseSsl, UseSSL, Ssl)
- Added 20+ comprehensive tests covering all edge cases
- Addressed code review comments: translated comment and improved null handling

### CODE REVIEW

**Comment by josemalm32** [comment]
@dotnet-policy-service agree

File: tests/CommunityToolkit.Aspire.Minio.Client.Tests/ConfigurationTests.cs [hunk 1]
```diff
@@ -12,4 +12,174 @@ public void EndpointIsNullByDefault() =>
...
+
+    [Fact]
+    public void ParseConnectionString_NullOrEmpty_DoesNotThrow()
+    {
+        var settings = new MinioClientSettings();
+        settings.ParseConnectionString(null);
+        Assert.Null(settings.Endpoint);
+        
+        settings = new MinioClientSettings();
+        settings.ParseConnectionString("");
```
Comment (Harold-Morgan) [inline tests/CommunityToolkit.Aspire.Minio.Client.Tests/ConfigurationTests.cs:96]: nit: string.Empty could be more convenient here

File: src/CommunityToolkit.Aspire.Minio.Client/MinioClientSettings.cs [hunk 2]
```diff
@@ -42,36 +44,82 @@ public sealed class MinioClientSettings
...
+        {
+            Credentials = new MinioCredentials
+            {
+                AccessKey = accessKey.ToString()!, SecretKey = secretKey.ToString()!
+            };
+        }
+    }
+
+    // Strict boolean parsing: only accepts bool or the strings "true"/"false" (case-insensitive).
+    private static bool TryParseBool(object? value, out bool result)
```
Comment (Harold-Morgan) [inline src/CommunityToolkit.Aspire.Minio.Client/MinioClientSettings.cs]: I think this method can be moved somewhere here https://github.com/CommunityToolkit/Aspire/tree/main/src/Shared

Here it seems a little bit out of place, despite being used only once :(

**Comment by Harold-Morgan** [comment]
PR itself looks good! 

I'm a little bit concerned about turning off security features by default, though, is it more convenient?

**Comment by josemalm32** [comment]
Hi @Harold-Morgan 

Thanks for review 💯 

If you use Minio locally, it is simpler to have SSL set to false. There is no need to configure self-signed certificates. And I suppose that if it is activated, the Hosting.MinIO package would have to be modified. The idea of adding SSL to the connection string is for production environments. In my case, I tried to connect but never succeeded because I always ended up calling port 80 (non-SSL) instead of 443 (SSL).

Thanks

File: src/CommunityToolkit.Aspire.Minio.Client/MinioClientSettings.cs [hunk 3]
```diff
@@ -12,6 +13,7 @@ public sealed class MinioClientSettings
     private const string ConnectionStringEndpoint = "Endpoint";
     private const string AccessKey = "AccessKey";
     private const string SecretKey = "SecretKey";
+    private static readonly string[] UseSslKeys = new[] { "UseSsl", "UseSSL", "Ssl" };
```
Comment (aaronpowell) [inline src/CommunityToolkit.Aspire.Minio.Client/MinioClientSettings.cs]: The `DbConnectionStringBuilder` is case insensitive when using `TryGetValue`: https://lab.razor.fyi/#bc69SgQxFAVgdrVwUy0-QZhqBiQws2VI4ywoYjGw_oCVSTYOkZl7l5tkfxALe99A8H0sfB9LWVRcxPKcjwOHvQwZawhb0r2w4fB5mIKHls82Ibpest0kpjpqUWPfI0jGlpq4DYYrDm7Fp6ZGAGejR5hF8tAeJ9_NHeUFf2Cjv6h4doohKm2sbJCiKqtqIi-DI6U1IchGh7BCmitt1uUP3eNvv8BUVpOMPUrGaoSAnRPX5KM79-ByG4y4oM2Ji1e6Sy7PFt-77Ihjinx7frmVopD_zL9Inu1RgpvRwcfr-9Pb3Xj_drAefAI

We should only care about a single key, `UseSsl`.

File: src/CommunityToolkit.Aspire.Minio.Client/MinioClientSettings.cs [hunk 4]
```diff
@@ -42,36 +44,82 @@ public sealed class MinioClientSettings
...
+            !string.IsNullOrEmpty(accessKey?.ToString()) && !string.IsNullOrEmpty(secretKey?.ToString()))
+        {
+            Credentials = new MinioCredentials
+            {
+                AccessKey = accessKey.ToString()!, SecretKey = secretKey.ToString()!
+            };
+        }
+    }
+
+    // Strict boolean parsing: only accepts bool or the strings "true"/"false" (case-insensitive).
```
Comment (aaronpowell) [inline src/CommunityToolkit.Aspire.Minio.Client/MinioClientSettings.cs]: Why? `bool.TryParse` is case insensitive so do we really care? This seems like code for codes sake.

File: src/CommunityToolkit.Aspire.Minio.Client/MinioClientSettings.cs [hunk 5]
```diff
@@ -42,37 +44,52 @@ public sealed class MinioClientSettings
...
+        
+        if (connectionBuilder.TryGetValue(AccessKey, out var accessKey)
+            &&
+            connectionBuilder.TryGetValue(SecretKey, out var secretKey)
+            && 
+            !string.IsNullOrEmpty(accessKey?.ToString()) && !string.IsNullOrEmpty(secretKey?.ToString()))
+            {
+            Credentials = new MinioCredentials
+            {
+                AccessKey = accessKey.ToString()!, SecretKey = secretKey.ToString()!
```
Comment (aaronpowell) [inline src/CommunityToolkit.Aspire.Minio.Client/MinioClientSettings.cs]: You can simplify this logic by using some pattern matching to avoid having to use `?` and `!`:

https://lab.razor.fyi/#hY5BSsRAEEUZ4kJ7JZ6gDDIkIH2ACdkko6KIDowouLI7dkJD0iXVHTWEHMO93sC1K7feSpIhEBRxVVT9x3_F3j3GVoQFiYpndu_Nq602Bawb61QVsenGl8IJnmJVoYkYexQEmZUQg1FPsJQpGqMyp9GsHWlTJLUu7xUFIbTwM4MY_GPEWESJoFj60EWM6RyCzEp-Rc2JcteirFXQU_4hYO2g9-WIIczn_QRtwW7KcsSB7pP9zY2f2ou6LC_pqHpwTTASIQOAHvvlSQRNPFLQ4JGCJh4p6B_PSIQhtIMqRWOxVPyGtFPn2qjgwM8RF9COL3V-GP2JSkELaMfWAe3OPKrN7c72x8vr51e-u3U3e559Aw

**Review by aaronpowell — CHANGES_REQUESTED** [review]

**Comment by josemalm32** [comment]
Adjusted with the required changes. 

Thanks 💯

**Comment by Harold-Morgan** [comment]
Some tests fail and need fixing :(

Apart from that looks great!

File: src/CommunityToolkit.Aspire.Minio.Client/MinioClientSettings.cs [hunk 6]
```diff
@@ -42,37 +44,54 @@ public sealed class MinioClientSettings
...
+            Uri.TryCreate(endpoint.ToString(), UriKind.Absolute, out var serviceUri))
+        {
+            Endpoint = serviceUri;
+        }
+        
+        // Check for UseSsl (and variants) in the connection string and parse it if present.
+            if (connectionBuilder.TryGetValue(UseSslKey, out var useSslObj))
             {
-                Endpoint = serviceUri;
+                if (bool.TryParse(useSslObj, out var parsed))
```
Comment (Harold-Morgan) [inline src/CommunityToolkit.Aspire.Minio.Client/MinioClientSettings.cs]: It seems that you need to explicitly cast useSslObj to string over here

**Comment by josemalm32** [comment]
Fix compilation error and improve readability of SSL parsing. 

Thanks to both of you😄

File: tests/CommunityToolkit.Aspire.Minio.Client.Tests/ConfigurationTests.cs [hunk 7]
```diff
@@ -12,4 +12,103 @@ public void EndpointIsNullByDefault() =>
...
+    {
+        var settings = new MinioClientSettings();
+        Assert.False(settings.UseSsl);
+    }
+
+    [Fact]
+    public void ParseConnectionString_HttpsUri_DoesNotInferUseSsl()
+    {
+        var settings = new MinioClientSettings();
+        settings.ParseConnectionString("https://minio.example.com:9000");
```
Comment (aaronpowell) [inline tests/CommunityToolkit.Aspire.Minio.Client.Tests/ConfigurationTests.cs:27]: Since the `ParseConnectionString` method is internal you'll need to expose it to the test project with `<InternalsVisibleTo Include="CommunityToolkit.Aspire.Minio.Client.Tests" />`

You'll find a bunch of examples of other projects using `InternalsVisibleTo` in the repo

**Review by aaronpowell — CHANGES_REQUESTED** [review]
Looks good and we're almost ready to merge

**Comment by aaronpowell** [comment]
@josemalm32 can you get this actioned in the next day so it can make the release this week with aspire 13

**Comment by josemalm32** [comment]
Hi @aaronpowell Sorry for the delay. With the latest change, the tests are working fine. I've tried running it locally and it works.

Thanks 😄
